internal/utils.go

```diff
@@ -3,47 +3,73 @@ package internal
 import (
 	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
-	"go.uber.org/zap"
 	"io"
 	"net/http"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Define static errors.
+var (
+	ErrJSONUnmarshalFailed  = errors.New("json.Unmarshal failed")
```

Comment:
You can also use `fmt.Errorf` instead of `errors.New` - both are ok, one has sprintf built in.